src/Debugger/Test/JsonTest.cs

```diff
@@ -41,12 +44,47 @@ public class JsonTest : IAsyncLifetime {
 
         [CompositeTest]
         [Category.R.Debugger]
-        [InlineData(@"'‘’“”'", @"""\u2018\u2019\u201c\u201d""")]
+        [InlineData("NULL", "null")]
+        [InlineData("NA", "null")]
+        [InlineData("NA_integer_", "null")]
+        [InlineData("NA_real_", "null")]
+        [InlineData("NA_character_", "null")]
+        [InlineData("TRUE[FALSE]", "null")]
+        [InlineData("0[FALSE]", "null")]
+        [InlineData("0L[FALSE]", "null")]
+        [InlineData("''[FALSE]", "null")]
+        [InlineData("TRUE", "true")]
+        [InlineData("FALSE", "false")]
+        [InlineData("0", "0")]
+        [InlineData("0L", "0")]
+        [InlineData(@"""text""", SameAsInput)]
+        [InlineData(@"""\r\n\t\f\""""", SameAsInput)]
+        [InlineData(@"""\a\v""", @"""\u0007\u000b""")]
+        [InlineData(@"""Ûñïçôδè ƭèжƭ: русский ελληνικά ქართული հայերեն हिन्दी ግዕዝ ᓀᐦᐃᔭᐍᐏᐣ 汉语/漢語 日本語 한국어/조선말 العَرَبِية‎ עברית""", SameAsInput)]
+        [InlineData(@"""‘’“”""", SameAsInput)]
+        [InlineData("list()", "[]")]
+        [InlineData("list(0, 's', NULL)", @"[0,""s"",null]")]
+        [InlineData("list(NA, 1, NA, 2, NA)", "[1,2]")]
+        [InlineData("structure(list(), names = ''[FALSE])", "{}")]
+        [InlineData("list(n = 0, s = 's', u = NULL)", @"{""n"":0,""s"":""s"",""u"":null}")]
+        [InlineData("list(n = 0, na = NA)", @"{""n"":0}")]
+        [InlineData("list(n = 0, na = NA)", @"{""n"":0}")]
+        [InlineData("as.environment(list())", "{}")]
+        [InlineData("as.environment(list(n = 0, s = 's', u = NULL))", @"{""n"":0,""s"":""s"",""u"":null}")]
+        [InlineData("as.environment(list(n = 0, na = NA))", @"{""n"":0}")]
+        [InlineData("as.environment(list(n = 0, na = NA))", @"{""n"":0}")]
+        [InlineData("list(as.environment(list(l = list())))", @"[{""l"":[]}]")]
```

Comment:
I suppose a proper fix would be to rewrite `to_json` so that it doesn't treat errors as fatal and just reports them. `host.cpp` can still treat an error that bubbles up as fatal at the point where it invokes it, but  invoking `toJSON` in R will then simply report the error as usual, and it can be used to test them.

Problem is, it's very tricky to write C++ code through which R errors can flow through. We have helpers for it that translate R errors to exceptions and vice versa; but because the former requires establishing a fresh R context, it's expensive to do in a tight loop, which is what JSON serializer is. I'll need to look closely at the code to see how this can be done safely.